frontend/tests/controllers/CoderOfTheMonthTest.php

```diff
@@ -570,4 +574,600 @@ public function testMentorCanChooseCoderOfTheMonth() {
+        );
+
+        // Now check if the other user has been saved on database too
+        $response = \OmegaUp\Controllers\User::apiCoderOfTheMonthList(new \OmegaUp\Request([
+            'auth_token' => $login->auth_token,
+            'date' => date('Y-m-d', \OmegaUp\Time::get()),
+            'category' => 'female'
+        ]));
+        //Now check if the third user has not participated in the coder of the month female
+        $this->assertFalse(
+            $extraIdentityMale->username ==
+            $response['coders'][0]['username']
+        );
+        $this->assertEquals(
+            $identity->username,
+            $response['coders'][0]['username']
+        );
+        $this->assertEquals(
+            $extraIdentity->username,
+            $response['coders'][1]['username']
+        );
+    }
+
+    public function testCoderOfTheMonthListFemale() {
+        ['user' => $user, 'identity' => $identity] = \OmegaUp\Test\Factories\User::createUser();
+        $auth_token = self::login($identity);
+
+        $r = new \OmegaUp\Request([
+            'auth_token' => $auth_token,
+            'category' => 'female'
+        ]);
+
+        $response = \OmegaUp\Controllers\User::apiCoderOfTheMonthList($r);
+
+        // Just one Coder of The Month, the one calculated on the previous test.
+        $this->assertCount(1, $response['coders']);
+
+        // Adding parameter date should return the same value, it helps
+        // to test getMonthlyList function.
+        // It should return two users (the ones that got stored on the previous test)
+        $r['date'] = date('Y-m-d', \OmegaUp\Time::get());
+        $r['category'] = 'female';
+        $response = \OmegaUp\Controllers\User::apiCoderOfTheMonthList($r);
+        $this->assertCount(2, $response['coders']);
+    }
+
+    public function testCodersOfTheMonthBySchoolFemale() {
+        ['user' => $user_1, 'identity' => $identity_1] = \OmegaUp\Test\Factories\User::createUser();
+        self::updateIdentity($identity_1, 'female');
+
+        ['user' => $user_2, 'identity' => $identity_2] = \OmegaUp\Test\Factories\User::createUser();
+        self::updateIdentity($identity_2, 'female');
+
+        // Identity 3 won't have school_id
+        ['user' => $user_3, 'identity' => $identity_3] = \OmegaUp\Test\Factories\User::createUser();
+        self::updateIdentity($identity_3, 'female');
+
+        // Add a custom school for identities 1 and 2
+        $school = SchoolsFactory::createSchool()['school'];
+
+        $login = self::login($identity_1);
+        \OmegaUp\Controllers\User::apiUpdate(new \OmegaUp\Request([
+            'auth_token' => $login->auth_token,
+            'school_id' => $school->school_id,
+        ]));
+
+        $login = self::login($identity_2);
+        \OmegaUp\Controllers\User::apiUpdate(new \OmegaUp\Request([
+            'auth_token' => $login->auth_token,
+            'school_id' => $school->school_id,
+        ]));
+
+        $today = date('Y-m') . '-01';
```

Comment:
nit:

```suggestion
        $today = date('Y-m-01', \OmegaUp\Time::get());
```

así evitas una concatenación. y el uso consistente de `\OmegaUp\Time::get()` evita que se creen cosas con fechas incorrectas.

igual en los demás lugares donde se usa ese patrón.

Reply:
listo